tests/include/mir/test/doubles/mock_drm.h

```diff
@@ -43,21 +44,37 @@ class FakeDRMResources
     int fd() const;
     int write_fd() const;
     drmModeRes* resources_ptr();
-
-    void add_crtc(uint32_t id, drmModeModeInfo mode);
-    void add_encoder(uint32_t encoder_id, uint32_t crtc_id, uint32_t possible_crtcs_mask);
+    drmModePlaneRes* plane_resources_ptr();
+    drmModeObjectProperties* get_object_properties(uint32_t id, uint32_t type);
+    drmModePropertyRes* get_property(uint32_t prop_id);
+    drmModePropertyBlobRes* get_property_blob(uint32_t prop_id);
+
+    uint32_t add_property(const char *name);
+    void add_crtc(uint32_t id, drmModeModeInfo mode,
+                  std::vector<uint32_t> prop_ids = {},
+                  std::vector<uint64_t> prop_values = {});
```

Comment:
There's a lot of passing around `std::vector<uint32_t> prop_ids, std::vector<uint64_t> prop_values` in this code which makes me wonder if there's an abstraction here with implicit invariants (like `prop_ids.size() == prop_values.size()`).

Reply:
Improved.